include/fastdds/dds/publisher/DataWriter.hpp

```diff
@@ -190,13 +199,44 @@ class DataWriter : public DomainEntity
             const fastrtps::rtps::InstanceHandle_t& handle);
      */
 
+    /**
+     * @brief This operation requests the middleware to delete the data (the actual deletion is postponed until there is no
+     * more use for that data in the whole system). In general, applications are made aware of the deletion by means of
+     * operations on the DataReader objects that already knew that instance. This operation does not modify the value of
+     * the instance. The instance parameter is passed just for the purposes of identifying the instance.
+     * When this operation is used, the Service will automatically supply the value of the source_timestamp that is made
+     * available to DataReader objects by means of the source_timestamp attribute inside the SampleInfo. The constraints
+     * on the values of the handle parameter and the corresponding error behavior are the same specified for the
+     * unregister_instance operation.
+     * @param data Pointer to data
+     * @param handle InstanceHandle of the data
+     * @return RETCODE_PRECONDITION_NOT_MET if the handle introduced does not match with the one associated to the data,
+     * RETCODE_OK if the data is correctly sent and RETCODE_ERROR otherwise.
+     */
     RTPS_DllAPI ReturnCode_t dispose(
             void* data,
             const fastrtps::rtps::InstanceHandle_t& handle);
 
+    /**
+     * @brief This operation requests the middleware to delete the data (the actual deletion is postponed until there is no
+     * more use for that data in the whole system). In general, applications are made aware of the deletion by means of
+     * operations on the DataReader objects that already knew that instance. This operation does not modify the value of
+     * the instance. The instance parameter is passed just for the purposes of identifying the instance.
+     * When this operation is used, the Service will automatically supply the value of the source_timestamp that is made
+     * available to DataReader objects by means of the source_timestamp attribute inside the SampleInfo. The constraints
+     * on the values of the handle parameter and the corresponding error behavior are the same specified for the
+     * unregister_instance operation.
+     * @param data Pointer to data.
+     * @return true if disposed, false if not.
+     */
     RTPS_DllAPI bool dispose(
```

Comment:
Should we change the return type of this to `ReturnCode_t` (perhaps on a different PR)?